pkg/cmd/server/api/types.go

```diff
@@ -759,29 +751,38 @@ type LDAPSchemaSpecificConfig struct {
 }
 
 type RFC2307Config struct {
-	// GroupQuery holds the template for an LDAP query that returns group entries
-	GroupQuery LDAPQuery
+	// AllGroupsQuery holds the template for an LDAP query that returns group entries.
+	AllGroupsQuery LDAPQuery
 
-	// GroupNameAttributes defines which attributes on an LDAP group entry will be interpreted as its' name
+	// GroupUIDAttributes defines which attribute on an LDAP group entry will be interpreted as its unique identifier.
+	// (ldapGroupUID)
+	GroupUIDAttribute string
+
+	// GroupNameAttributes defines which attributes on an LDAP group entry will be interpreted as its name to use for
+	// an OpenShift group
 	GroupNameAttributes []string
 
-	// GroupMembershipAttributes defines which attributes on an LDAP group entry will be interpreted
-	// as its' members
+	// GroupMembershipAttributes defines which attributes on an LDAP group entry will be interpreted  as its members.
+	// The values contained in this attribute must be substitutable into your UserQuery.QueryAttribute
 	GroupMembershipAttributes []string
 
-	// UserQuery holds the template for an LDAP query that returns user entries
-	UserQuery LDAPQuery
+	// AllUsersQuery holds the template for an LDAP query that returns user entries.
+	AllUsersQuery LDAPQuery
+
+	// UserUIDAttribute defines which attribute on an LDAP user entry will be interpreted as its unique identifier.
+	// It must correspond to values that will be found from the GroupMembershipAttributes
+	UserUIDAttribute string
 
-	// UserNameAttributes defines which attributes on an LDAP user entry will be interpreted as its' name
+	// UserNameAttributes defines which attributes on an LDAP user entry will be interpreted as its OpenShift user name.
```

Comment:
...will be used, in order, as its OpenShift user name. The first attribute with a non-empty value is used.